File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/LocationInfoData.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import dev.jefy.connectpro.portfolio.domain.vo.LocationInfo;

/**
 * @author Jôph Yamba
 */
public record LocationInfoData(
        String country,
        String city,
        String address,
        Double latitude,
        Double longitude
){
    public static LocationInfoData from(LocationInfo locationInfo) {
        return new LocationInfoData(
                locationInfo.country(), 
                locationInfo.city(), 
                locationInfo.address(), 
                locationInfo.latitude(), 
                locationInfo.longitude()
        );
    }

    public LocationInfo toDomain(){
        return new LocationInfo(
                this.country,
                this.city,
                this.address,
                this.latitude,
                this.longitude
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/LocationInfoData.kt
package dev.jefy.connectpro.portfolio.application.dtos

import dev.jefy.connectpro.portfolio.domain.vo.LocationInfo


/**
 * @author Jôph Yamba
 */

data class LocationInfoData(
    val country: String,
    val city: String,
    val address: String?,
    val latitude: Double?,
    val longitude: Double?
) {
    fun toDomain(): LocationInfo {
        return LocationInfo(
            country = this.country,
            city = this.city,
            address = this.address,
            latitude = this.latitude,
            longitude = this.longitude
        )
    }
    
}

fun LocationInfo.toData(): LocationInfoData = LocationInfoData(
    country = this.country,
    city = this.city,
    address = this.address,
    latitude = this.latitude,
    longitude = this.longitude
)
